clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -3118,11 +3115,29 @@ raw2trace_t::write(raw2trace_thread_data_t *tdata, const trace_entry_t *start,
                         (tdata->chunk_count_ - 1) * chunk_instr_count_ +
                         tdata->cur_chunk_instr_count;
                     tdata->last_cpu_ = static_cast<uint>(it->addr);
-                    tdata->sched.emplace_back(tdata->tid, tdata->last_timestamp_,
-                                              tdata->last_cpu_, instr_count);
-                    tdata->cpu2sched[it->addr].emplace_back(
-                        tdata->tid, tdata->last_timestamp_, tdata->last_cpu_,
-                        instr_count);
+                    // Avoid identical entries, which are common with the end of the
+                    // previous buffer's timestamp followed by the start of the next.
+                    if (tdata->sched.empty() ||
```

Comment:
To help with readability of this if condition: could construct an element and check `==` with `back()`?

Reply:
I started on that with an operator== but balked b/c of the extra object creation cost...but maybe it's fine, it's a small struct.  Done, made an operator!=.